src/hooks/useScroll.js

```diff
@@ -3,15 +3,32 @@ import { useKeys } from 'hooks'
 export const useScroll = (
   elementRef,
   step,
-  axis
+  axis,
+  event
 ) => {
   const prop = axis === 'x' ? 'scrollLeft' : 'scrollTop'
   useKeys({
     ArrowDown: () => {
+      const previous = elementRef.current[prop]
       elementRef.current[prop] += step
+      const after = elementRef.current[prop]
+
+      // show the next page (section in article)
+      if (previous === after) {
+        event.nextPage()
+        elementRef.current[prop] = 0
+      }
     },
     ArrowUp: () => {
+      const previous = elementRef.current[prop]
       elementRef.current[prop] -= step
+      const after = elementRef.current[prop]
+
+      // show the previous page (section in article)
+      if (previous === after) {
+        event.prevPage()
+        elementRef.current[prop] = 0
```

Comment:
_here we show the location 0, as we don't have previous section information (can only know this after the section is being rendered)._

Reply:
We'll have to solve that. It's really confusing.

Reply:
There is a quick and dirty solution for this

```js
setTimeout(() => {
  const scrollWidth = elementRef.current.scrollWidth
  const offset = 240
  elementRef.current[prop] = scrollWidth - offset
}, 0)
```

The real challenging we have here is that we render section by section, we don't have the previous section in the view, therefore no scrollWidth information. The current workflow is 

1. Render the section 
2. set the current scrollLeft

Therefore, this quick solution, you will see the first page of the section, then immediately to the last page of that section. 

Or we might need to use an extra layer (loading) when doing the transition.

Reply:
If we do the second step inside `useLayoutEffect` instead of `useEffect` there will be no flash between the steps.

Reply:
Thanks!! Let me try that.